share/new_eds/in_mem.go

```diff
@@ -12,29 +12,30 @@ import (
 	"github.com/celestiaorg/celestia-node/share/shwap"
 )
 
-var _ EdsFile = (*MemFile)(nil)
+var _ EDS = InMem{}
 
-type MemFile struct {
-	Eds *rsmt2d.ExtendedDataSquare
+// InMem is an in-memory implementation of EDS.
+type InMem struct {
+	*rsmt2d.ExtendedDataSquare
```

Comment:
Why embedded?

Reply:
To allow access to inherited type

Reply:
It is meant to extend imported  rsmt2d.ExtendedDataSquare type with few more methods